beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -198,7 +197,7 @@ func (s *Service) validateBeaconBlock(ctx context.Context, blk interfaces.Signed
 		return err
 	}
 
-	if err := blocks.VerifyBlockSignature(parentState, blk.Block().ProposerIndex(), blk.Signature(), blk.Block().HashTreeRoot); err != nil {
+	if err := blocks.VerifyBlockSignatureUsingCurrentFork(parentState, blk); err != nil {
```

Comment:
I'm concerned that this needed to be changed. 

The `parentState` should have the correct `fork` object, if not, then the bug is somewhere else